client/src/components/Public/EntitySearchResult/EntitySearchResult.js

```diff
@@ -1,43 +1,84 @@
 // @flow
-import React from 'react'
+import React, {PureComponent} from 'react'
 import {connect} from 'react-redux'
 import {compose} from 'redux'
 import {branch} from 'recompose'
 import {withDataProviders} from 'data-provider'
-import {entitySearchForSelector, entitySearchEidsSelector} from '../../../selectors'
-import {entitySearchProvider, entityDetailProvider} from '../../../dataProviders/sharedDataProviders'
+import {pick, uniqWith, isEqual} from 'lodash'
+import {
+  entitySearchForSelector,
+  entitySearchEidsSelector,
+  entitiesToHighlightSelector,
+} from '../../../selectors'
+import {
+  entitySearchProvider,
+  entityDetailProvider,
+} from '../../../dataProviders/sharedDataProviders'
+import {makeLocationsSelected, zoomToLocation} from '../../../actions/publicActions'
 import EntitySearchResultItem from '../EntitySearchResultItem/EntitySearchResultItem'
+import {getBoundsFromLocations} from '../../../services/map'
+import {NAVBAR_HEIGHT} from '../../../constants'
+import type {NewEntityDetail, Center, State} from '../../../state'
+import type {ObjectMap} from '../../../types/commonTypes'
 
 type Props = {
   searchFor: string,
-  entitySearchEids: Array<number>,
+  entityDetails: ObjectMap<NewEntityDetail>,
+  makeLocationsSelected: (Center[]) => void,
+  zoomToLocation: (Center, number) => void,
 }
 
-const EntitySearchResult = ({entitySearchEids}: Props) => (
-  entitySearchEids.map((eid) =>
-    <EntitySearchResultItem key={eid} eid={eid} />
-  )
-)
+class EntitySearchResult extends PureComponent<Props> {
+  componentDidMount() {
+    this.highlightLocations()
+  }
+
+  componentDidUpdate(oldProps) {
+    if (this.props.searchFor !== oldProps.searchFor) {
+      this.highlightLocations()
+    }
+  }
+
+  highlightLocations = () => {
+    const {makeLocationsSelected, entityDetails, zoomToLocation} = this.props
+    const locations = uniqWith(
+      Object.values(entityDetails).map((detail) => pick(detail, ['lat', 'lng', 'address_id'])),
```

Comment:
No need to use Object.values if you use map from lodash.

Reply:
True, but I prefer using native `map` for performance reasons :slightly_smiling_face: